subiquitycore/view.py

```diff
@@ -82,6 +86,42 @@ def show_stretchy_overlay(self, stretchy):
         stretchy.opened()
         self._w = StretchyOverlay(disabled(self._w), stretchy)
 
+    async def ask_confirmation(self, title: str, question: str,
+                               confirm_label: str, cancel_label: str) -> bool:
+        """ Open a confirmation dialog using a strechy overlay.
+        If the user selects the "yes" button, the function returns True.
+        If the user selects the "no" button or closes the dialog, the function
+        returns False.
+        """
+        confirm_event = asyncio.Event()
```

Comment:
would it be slightly less clunky to use a queue for this?

Reply:
Thanks for the suggestion. Yes that seems cleaner!